libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1629,9 +1629,20 @@ bool TypeChecker::visit(FunctionCall const& _functionCall)
 			auto const& argType = type(*arguments[i]);
 			if (functionType->takesArbitraryParameters())
 			{
+				bool errored = false;
 				if (auto t = dynamic_cast<RationalNumberType const*>(argType.get()))
 					if (!t->mobileType())
+					{
 						m_errorReporter.typeError(arguments[i]->location(), "Invalid rational number (too large or division by zero).");
+						errored = true;
+					}
+				if (!errored && !(
```

Comment:
This is a bit hard to read. Basically it allows any basic type except structs. Is there any way to make this nicer?

Reply:
This is the same check that is present in the encoder. It also checks a bit more than just structs, because not all types have mobile types or encoding types.

Reply:
That's what I meant, with "Basically it allows any basic type except structs.".

But is there a way to make that easier to read? Ah well, it can be improved later on.